test/integration/core/el-rewards.integration.ts

```diff
@@ -0,0 +1,92 @@
+import { expect } from "chai";
+import { ContractTransactionReceipt } from "ethers";
+import { ethers } from "hardhat";
+
+import { HardhatEthersSigner } from "@nomicfoundation/hardhat-ethers/signers";
+
+import { ether } from "lib";
+import { getProtocolContext, ProtocolContext } from "lib/protocol";
+
+import { Snapshot } from "test/suite";
+
+describe("EL rewards distribution", () => {
+  let ctx: ProtocolContext;
+  let snapshot: string;
+  let stranger: HardhatEthersSigner;
+
+  const REWARD_AMOUNT = ether("1");
+
+  before(async () => {
+    ctx = await getProtocolContext();
+    [stranger] = await ethers.getSigners();
+    snapshot = await Snapshot.take();
+  });
+
+  after(async () => await Snapshot.restore(snapshot));
```

Comment:
```suggestion
  beforeEach(bailOnFailure);

  after(async () => await Snapshot.restore(snapshot));
```

Not to continue scenario test if one of the tests failed